src/main/java/com/thoughtworks/gauge/TableRow.java

```diff
@@ -84,12 +84,42 @@ public String toString() {
     public List<TableCell> getTableCells() {
         List<TableCell> listOfCells = new ArrayList<TableCell>();
 
-        //Since we have a LinkedHashMap, the order of entrySet() is guaranteed.
+        // Since we have a LinkedHashMap, the order of entrySet() is guaranteed.
         for (Map.Entry<String, String> mapEntry : cells.entrySet()) {
             TableCell cell = new TableCell(mapEntry.getKey(), mapEntry.getValue());
             listOfCells.add(cell);
         }
 
         return listOfCells;
     }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
+        int result = 1;
+        result = prime * result + ((cells == null) ? 0 : cells.hashCode());
+        return result;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null) {
+            return false;
+        }
+        if (getClass() != obj.getClass()) {
+            return false;
+        }
+        TableRow other = (TableRow) obj;
+        if (cells == null) {
```

Comment:
This can be simplified to `return cells == null ? other.cells == null : cells.equals(other.cells);`